kornia/contrib/object_detection.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+from kornia.core import Module, Tensor, concatenate, tensor
+
+
+class ObjectDetector(Module):
+    def __init__(self, model: Module, confidence_threshold: float = 0.3, postprocess_type: str = "detr"):
+        if postprocess_type.lower() != "detr":
+            raise ValueError("Currently only DETR post-process is supported")
+        super().__init__()
+        self.model = model
+        self.confidence_threshold = confidence_threshold
+        self.postprocess_type = postprocess_type
+
+    def postprocess(self, data: dict[str, Tensor], height: int, width: int) -> list[Tensor]:
```

Comment:
What if this this a class ? Think of being configurable e.g. in case on inference vs train ?

Reply:
I have no opinion about Object Detection API design. This is just a simple API with only RT-DETR in mind. I think if we prioritise merging this PR, the details of the Object Detection API can be re-done in a separate PR

Here are some of my thoughts:

- If postprocess is a class, why don't we just include it directly inside `RTDETR`?
- Training highly depends on the model. For example, for RT-DETR, during training, the score head outputs more stuff to be used for training (to be specific, score head for each DETR decoder layer), while during inference, only the last decoder layer is used. There are definitely more training details for RT-DETR that I have yet looked into it.
- Perhaps to keep it simple, the model will return training loss if it is in training mode, while returning raw model outputs (or with postprocessing?) in eval mode.
- For training, I think it's easier if we use the network class directly i.e. `RTDETR` instead of through the wrapper `ObjectDetector`. I think the class `ObjectDetector` should only be used for inference.